### PR TITLE
[#13] Create a manager registry in order to get access to all managers

### PR DESCRIPTION
**Pull request by StephaneCapponi**

### CODE REVIEW

**Comment by StephaneCapponi** [comment]
@JRbemt, could you check that pull request ?

File: docs/how-to/custom-managers-and-hooks.txt
Comment (JRbemt) [inline docs/how-to/custom-managers-and-hooks.txt:1]: One ref to :class:`~socon.core.manager.BaseManager._managers`

File: docs/intro/tutorials/3_manager.txt
Comment (JRbemt) [inline docs/intro/tutorials/3_manager.txt:1]: :class:`~socon.core.manager.BaseManager` outdated reference on line 182

File: docs/ref/manager.txt [hunk 1]
```diff
@@ -9,6 +9,30 @@ on how to use them and how to write your own managers and hooks,
...
 You can also find a tutorial about managers and hooks :doc:`here </intro/tutorials/2_change_behavior>`.
 
+Manager Registry
+================
+
+.. class:: ManagerRegistry
+
+    Base class that stores all the managers. At start Socon creates a
+    ``managers`` object that is accessible like that::
+
```
Comment (JRbemt) [inline docs/ref/manager.txt:19]: At the start Socon creates a ``managers`` object that is accessible like this::

File: socon/core/manager.py [hunk 2]
```diff
@@ -20,6 +20,45 @@
     from socon.core.registry.config import RegistryConfig
 
 
+__all__ = ["managers", "BaseManager", "Hook"]
+
+
+class ManagerRegistry:
+    """Base class that register all managers"""
```
Comment (JRbemt) [inline socon/core/manager.py:27]:
```suggestion
    """Base class that registers all managers"""
```

Comment (JRbemt) [inline socon/core/manager.py:27]:
```suggestion
    """Base class that registers all managers"""
```

**Review by JRbemt — CHANGES_REQUESTED** [review]
Looks good, some references to BaseManager should be updated in the docs. But the code looks fine

**Review by JRbemt — APPROVED** [review]

**Pull request merged**